addons/survey/models/survey_survey.py

```diff
@@ -434,6 +467,75 @@ def get_input_summary(self, question, current_filters=None):
 
         return result
 
+    # Private Methods
+    # TODO DBE : Reorganise this class to have public methods in public method section and private in private one
+    def _create_certification_badge_trigger(self):
+        self.ensure_one()
+        goal = self.env['gamification.goal.definition'].create({
+            'name': self.title,
+            'description': "%s certification passed" % self.title,
+            'domain': "['&', ('survey_id', '=', %s), ('quizz_passed', '=', True)]" % self.id,
+            'computation_mode': 'count',
+            'display_mode': 'boolean',
+            'model_id': self.env.ref('survey.model_survey_user_input').id,
+            'condition': 'higher',
+            'batch_mode': True,
+            'batch_distinctive_field': self.env.ref('survey.field_survey_user_input__partner_id').id,
+            'batch_user_expression': 'user.partner_id.id'
+        })
+        challenge = self.env['gamification.challenge'].create({
+            'name': '%s challenge certificate' % self.title,
+            'reward_id': self.certification_badge_id.id,
+            'state': 'inprogress',
+            'period': 'once',
+            'category': 'certification',
+            'reward_realtime': True,
+            'report_message_frequency': 'never',
+            'user_domain': [('karma', '>', 0)],
+            'visibility_mode': 'personal'
+        })
+        self.env['gamification.challenge.line'].create({
+            'definition_id': goal.id,
+            'challenge_id': challenge.id,
+            'target_goal': 1
+        })
+
+    def _handle_certification_badges(self, vals):
+        if vals.get('certification_give_badge'):
+            # If no badge set on records
+            surveys_without_badge = self.filtered(lambda s: not s.certification_badge_id)
+            if surveys_without_badge:  # first validation (need write to be done before creating badge trigger)
+                if not vals.get('certification_badge_id'):
+                    raise UserError('Certification badge must be configured if Give Badge is set.')
+                if len(surveys_without_badge) > 1:
+                    raise UserError('Same certification badge cannot be configured on multiple certification surveys.')
+            # If badge already set on records, reactivate them all.
+            surveys_with_badge = (self - surveys_without_badge).filtered(lambda survey: not survey.certification_badge_id.active)
+            surveys_with_badge.mapped('certification_badge_id').write({'active': True})
+
+            result = super(Survey, self).write(vals)
+
+            for survey in (surveys_with_badge + surveys_without_badge):
+                survey._create_certification_badge_trigger()
+        else:
```

Comment:
If not certification_give_badge . Here normally we can put super back to write() . Not sure we should handle badge deletion / archive depending on owners; maybe it is simpler to always archive. I let you choose, I would just be very very sad.